src/metabase/models/collection.clj

```diff
@@ -1084,14 +1089,17 @@
 
 (defn personal-collection-with-ui-details
   "For Personal collection, we make sure the collection's name and slug is translated to user's locale
-  This is only used for displaying purposes, For insertion or updating  the name, use site's locale instead"
+  This is only used for displaying purposes, For insertion or updating  the name, use site's locale instead.
+  We also add a `personal_owner_active` boolean flag indicating whether the owner of the collection is an active user,
+  since only personal collections of deactivated users can be archived or unarchived."
   [{:keys [personal_owner_id] :as collection}]
   (if-not personal_owner_id
     collection
     (let [collection-name (user->personal-collection-name personal_owner_id :user)]
       (assoc collection
-             :name collection-name
-             :slug (u/slugify collection-name)))))
+             :name                  collection-name
+             :slug                  (u/slugify collection-name)
+             :personal_owner_active (t2/select-one-fn :is_active 'User :id personal_owner_id)))))
```

Comment:
```suggestion
             :owner_is_active (t2/select-one-fn :is_active 'User :id personal_owner_id)))))
```

I don't think I understand the original name, `owner_is_active` or `is_owner_active` feels more natural to me.

Reply:
I was mirroring `personal_owner_id` which is already a field on collections. But I'm fine with changing it to `owner_is_active`